### DISCUSSION TITLE
Encoding graph elements?

### DISCUSSION THREAD

**Discussion by Z-Rajaei** [networkx/networkx#7340] [Q&A]
Hello,
I have some problems using NetworkX.
I have some nodes, each node has some attributes and attribute values (For example, name = "Jame").
When we create a graph using the following codes, shouldn't we encode the attribute value ("green") to numbers before assigning it to the node?
I want to encode values. What encodings can I use?
Should the name of attributes encoded too?

G=nx.Graph()
 G.add_nodes_from([  (4, {"color": "red"}), (5, {"color": "green"})

**Comment by MridulS** [comment]
>When we create a graph using the following codes, shouldn't we encode the attribute value ("green") to numbers before assigning it to the node?

It's just a python dictionary underneath, so you are free to use anything hashable as the name of the attribute and it can hold strings as the value just fine. Depends on what you want to do with the attributes.


>  I want to encode values. What encodings can I use?

I guess any mapping-like thing would work here? What is the use case?

Something like
``` python
color_map = {'red': 1, 'green': 2, 'black': 3}
def encode(color):
    return color_map[color]
```
would work just fine?

**Reply by Z-Rajaei** [reply]
Thank you.
If I have attributes like name="Jane", Age =12 and Description="He is a boy works at AB company".
How can it be encoded?
Can I use Word2vec encoding for the attribute names and values?

**Reply by rossbar** [reply]
I think there's too much focus on the word "encode" here - there is no requirement that you represent anything as a feature vector or anything like that. As @MridulS stated, you can store any hashable object as an attribute of either a node or an edge - just as you could with a standard Python dictionary. How you do this for your *specific* problem is left entirely up to you!
